### ISSUE TITLE
Padding content affects the result.

### ISSUE THREAD

**Issue by MashPlant** [simdjson/simdjson#1834]
**Describe the bug**
I cannot find a clear description of the content requirements of the padding area from the public documentation, and from the code comment I see that its content should not affect the result.

https://github.com/simdjson/simdjson/blob/8c1bfe782b718d128fd655f2070971b52abfc9b0/include/simdjson/generic/ondemand/parser.h#L78-L81

However, I found that under certain circumstances the contents in the padding area can affect the result. Currently, I only found that some content can make illegal inputs produce normal results instead of the expected error.

**To Reproduce**
```c++
#include "simdjson.h"
#include <iostream>

using namespace simdjson;

int main() {
  ondemand::parser parser;
  auto json = "[[]"_padded;
  //  json.data()[json.size()] = ']';
  auto doc = parser.iterate(json);
  size_t cnt = doc.count_elements();
  std::cout << cnt;
}
```

The expected behavior of the code is throwing an exception since `"[[]"` is not a valid JSON array. However, if the commented line is uncommented, the code will print `1`, which is the length of `"[[]]"`.

**simjson release**

v1.1.0 (commit 8c1bfe782b718d128fd655f2070971b52abfc9b0).

**Configuration (please complete the following information if relevant)**
 - OS: macOS Monterey
 - Compiler Apple clang version 13.0.0 (clang-1300.0.29.3) x86_64-apple-darwin21.4.0
 - Version: 12.3.1
 - Optimization setting: None

**Comment by lemire** [comment]
Indeed, something is wrong. We should be checking for this case already. In fact, we do related checks.

**Comment by lemire** [comment]
I will work on this soon.

Congratulations on finding this bug.

cc @jkeiser
